### ISSUE TITLE
Visual feedback long tap before vs after release

### ISSUE THREAD

**Issue by kaoneko** [micku7zu/QuickCursor#19]
So I'm using long tap (on the tracker) to trigger the long tap action, and I just updated from version 1.19.4 to the latest version.

Now in 1.19.4 when you held your finger on the tracker, once the long tap time threshold had been reached the cursor would show the visual feedback (ripple), so you knew you could now release the tracker and it would trigger the associated long tap action. This was true for vibration on long tap action as well by the way, it was felt _before_ release.

In the latest version, the visual feedback (and possibly vibration as well, I have vibration turned off nowadays) is only shown _after_ releasing the tracker, so you're unsure when you'll trigger a long tap on the tracker instead of a single tap. This is at least true when the associated action is the long tap action.

It seems we went from feedback about the long tap being triggered (when long tap could only be associated with long tap or gesture recording) to feedback about the associated action being triggered. I think we need both, so both the old and the new way are incomplete.

Indicating a long tap will be triggered instead of a single tap could for example be done by one of the following:
- filling up the cursor from the center to the outside until the threshold is reached (which is 500 ms by default, by the way)
- changing the color of the cursor once it reaches the threshold or transitioning it to the new color or tint as it approaches the threshold

Or is all this already possible and have I just missed a toggle somewhere?

**Comment by micku7zu** [comment]
In version 1.21.4 I've added more options to choose on "tracker long tap" and I had to make them all work mostly the same.

I've choosen the **visual feedback to happen on action trigger** (on gesture simulation or some action at cursor position, in this case long tap, or swipe, or copy text at cursor position, etc) and the **vibration feedback when the action was registered** (when the time needed to trigger the action has passed).

I understand that this is not enough for anyone and some users, like in your case, might want the visual feedback to see when the action was registered and maybe the vibration when the action was triggered, but right now this is the way it works.

**Thanks for feedback!** I will see what other feedback I will get about this one and see if it is worth to extend it more.

Adding more and more choices might be good for some users but will also increase the complexity of the app (to configure it and to maintain it).

It looks like there is a bug in version 1.24.3 with the "Vibration feedback" setting. The change is not propagated only after you change "long tap action" again. So, if you want to activate vibration, you have to toggle it on, and then change "long tap action". I've fixed it in 1.24.4, and I will release it as soon as possible.

**Comment by micku7zu** [comment]
> It looks like there is a bug in version 1.24.3 with the "Vibration feedback" setting. The change is not propagated only after you change "long tap action" again. So, if you want to activate vibration, you have to toggle it on, and then change "long tap action". I've fixed it in 1.24.4, and I will release it as soon as possible.

Fixed in version [1.24.4](https://github.com/micku7zu/QuickCursor/releases/tag/199): https://github.com/micku7zu/QuickCursor/releases/tag/199

**Comment by kaoneko** [comment]
So you did carefully consider the whole scenario! It's just that when someone has vibrations turned off they're left with only the visual feedback. And my current ROM (Google Pixel Android 13) doesn't allow me to disable general haptics in Androids UI separately from haptics in third party apps. It's super stupid. I don't want haptic feedback when opening quick settings or flicking through my recent apps, but I do want third party apps to be able to provide haptic feedback. But that's not possible with Google's flavour of Android, at least not on my phone ¯\\\_(ツ)\_/¯ I think on LineageOS this was not an issue.

I'll stick with the older version of Quick Cursor for now.

**Comment by micku7zu** [comment]
> And my current ROM (Google Pixel Android 13) doesn't allow me to disable general haptics in Android's UI separately from haptics in third party apps.

This is really strange. I never heard about this situation. It sounds like a bug on Pixel, not a feature.

**Comment by kaoneko** [comment]
Yes. Apparently this behaviour [was introduced with Android 13](https://www.reddit.com/r/GooglePixel/comments/wp9vn8/gboard_haptic_feedback_stopped_working_after/ikfh50j/).
